tomoima525/zkp-toy
# Deploy ropsten

# Preparation
- Setup an account on Alchemy
- Set ROPSTEN_URL on .env
- Export the private key from Metamask
  - open Metamask and go to Account Details > Export Private Key
  - copy and paste it in PRIVATE_KEY of .env file

# Deployment
- Run the command below

```
$ npx hardhat run scripts/deploy.ts --network ropsten
No need to generate any newer typings.
Deploying contracts with the account: 0xa62d3D7334c3d47616DD5FF665C6Bdc433F85002
Account balance: 300000000000000000
Nothing to compile
No need to generate any newer typings.
VoteEvenOrOdd deployed to: 0xfe6249B1998416737EB8898000aaDcb099C38cC3
```

# Verification
- Generate API Key at etherscan.io
- Setup etherscan api key inside .env
- Run the command below(Deployed contract address is taken from the result above)

```
$ npx hardhat verify --network ropsten 0xfe6249B1998416737EB8898000aaDcb099C38cC3                
Nothing to compile
No need to generate any newer typings.
Compiling 1 file with 0.8.4
Warning: SPDX license identifier not provided in source file. Before publishing, consider adding a comment containing "SPDX-License-Identifier: <SPDX-License>" to each source file. Use "SPDX-License-Identifier: UNLICENSED" for non-open-source code. Please see https://spdx.org for more information.
--> contracts/circuits/VoteEvenOrOdd.sol


Successfully submitted source code for contract
contracts/circuits/VoteEvenOrOdd.sol:VoteEvenOrOdd at 0xfe6249B1998416737EB8898000aaDcb099C38cC3
for verification on Etherscan. Waiting for verification result...

Successfully verified contract VoteEvenOrOdd on Etherscan.
https://ropsten.etherscan.io/address/0xfe6249B1998416737EB8898000aaDcb099C38cC3#code
```

https://ropsten.etherscan.io/address/0xfe6249B1998416737EB8898000aaDcb099C38cC3#code
